backend/src/services/base.py

```diff
@@ -11,12 +18,12 @@ def __init__(self, session: AsyncSession, config: ControllerConfig | None = None
         self.config = config
 
     async def get_paginated_response(
-        self, model, stmt: Select, query_parameters: QueryParameters
+        self, model, stmt: Select, query_parameters: QueryParameters, schema: Type[GenericBaseModel]
```

Comment:
Read this link -> [link](https://docs.python.org/3/library/typing.html#:~:text=Generic%20classes%20implicitly%20inherit%20from%20Generic.%20For%20compatibility%20with%20Python%203.11%20and%20lower%2C%20it%20is%20also%20possible%20to%20inherit%20explicitly%20from%20Generic%20to%20indicate%20a%20generic%20class%3A)

```
from typing import Generic

class BaseService(Generic[GenericBaseModel]):
     ....
```